velox/docs/develop/anti-join.rst

```diff
@@ -0,0 +1,576 @@
+
+	  id  | value
+	------+-------
+	 NULL |     0
+	    2 |     2
+	    3 |     3
+	(3 rows)
+
+	> WITH
+		t as (SELECT * FROM unnest(array[null, 1, 2], array[0, 1, 2]) as _t(id, value)),
+		u as (SELECT * FROM unnest(array[null, 2, 3], array[0, 1, 2]) as _t(id, value))
+	SELECT * FROM t FULL JOIN u ON t.id = u.id;
+
+	  id  | value |  id  | value
+	------+-------+------+-------
+	    1 |     1 | NULL | NULL
+	    2 |     2 |    2 |     1
+	 NULL | NULL  |    3 |     2
+	 NULL |     0 | NULL | NULL
+	 NULL | NULL  | NULL |     0
+	(5 rows)
+
+
+NOT IN <subquery> Semantics
+---------------------------
+
+We need to consider 3 cases of NOT IN queries:
+
+* Subquery returns a row with a NULL.
+* Subquery returns no rows.
+* Subquery returns one or more rows and no row has a NULL.
+
+NOT IN with NULLs in the subquery
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Consider NOT IN query:
+
+.. code-block:: sql
+
+	SELECT * FROM t WHERE t.id NOT IN (SELECT * FROM u)
+
+This query returns an empty result.
+
+.. code-block:: sql
+
+	> WITH
+	     t as (SELECT * FROM unnest(array[null, 1, 2], array[0, 1, 2]) as _t(id, value)),
+	     u as (SELECT * FROM unnest(array[null, 2, 3], array[0, 2, 3]) as _t(id, value))
+	SELECT * from t WHERE t.id NOT IN (SELECT id FROM u);
+
+	 id | value
+	----+-------
+	(0 rows)
+
+
+This is because the IN LIST contains 3 values: NULL, 2, 3. In SQL, NULL is
+considered an unknown value. In this case the IN LIST contains unknown value
+and we cannot definitively say whether any given value is in the list or not.
+Therefore NOT IN predicate returns a NULL (unknown), hence, the query returns
+no result. You can use the following queries to confirm the semantics of the
+NOT IN predicate.
+
+.. code-block:: sql
+
+	> SELECT 1 not in (null, 2, 3);
+
+	 _col0
+	-------
+	 NULL
+	(1 row)
+
+	> SELECT null not in (null, 2, 3);
+
+	 _col0
+	-------
+	 NULL
+	(1 row)
+
+NOT IN without NULLs in the subquery
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now, consider NOT IN query where subquery doesn’t return NULLs (by removing
+NULLs from the u table or adding u.id IS NOT NULL predicate to the subquery).
+
+
+.. code-block:: sql
+
+	SELECT * FROM t WHERE t.id NOT IN (
+		SELECT * FROM u WHERE u.id IS NOT NULL
+	)
+
+This query returns a single row with id 1.
+
+.. code-block:: sql
+
+	> WITH
+	     t as (SELECT * FROM unnest(array[null, 1, 2], array[0, 1, 2]) as _t(id, value)),
+	     u as (SELECT * FROM unnest(array[2, 3], array[1, 2]) as _t(id, value))
+	SELECT * from t WHERE t.id NOT IN (SELECT id FROM u);
+
+	 id | value
+	----+-------
+	  1 |     1
+	(1 row)
+
+In this case the IN LIST contains 2 values: 2 and 3. NULL NOT IN (2, 3) returns
+NULL because we cannot tell definitively whether an unknown value is part of a
+set or not, hence, not included in the results. 1 NOT IN (2, 3) returns true,
+hence, included in the results. 2 NOT IN (2, 3) returns false, hence, not
+included in the results. NOT IN with empty subquery Now, consider a NOT IN
```

Comment:
"NOT IN with empty subquery" should be an header here

Reply:
That's a good catch. Thanks. Fixed.